connect/runtime/src/test/java/org/apache/kafka/connect/runtime/AbstractHerderTest.java

```diff
@@ -299,6 +307,46 @@ public void testConfigValidationTransformsExtendResults() {
         verifyAll();
     }
 
+    @Test()
+    public void testConfigValidationPrincipalOnlyOverride() {
```

Comment:
Why are we just testing the `Principal` policy? Should we also test the other built-in policies to make sure they're instantiated and used properly?

Reply:
The individual policies have their own unit tests. Also, in these tests, we are explicitly creating an instance of the policy and passing it around.

Reply:
ok